frontend-chat/lib/core.js

```
@@ -0,0 +1,45 @@
+var $ = require('jquery');
+var angular = require('angular');
+
+var app = angular.module('chatApp', []);
+app.controller('chatCtrl', ['$scope', 'dataService', function($scope, dataService){
+	$scope.lastMessage;
+	$scope.messages;
+	$scope.newMessage;
+
+	getRecentMessages();
+
+	function getRecentMessages(){
+		dataService.getRecentMessages().then(function(result){	
```

You should handle errors from the request.

On top of that, you should call `.done()` at the very end of the promise chain to ensure that other errors are not swallowed.